src/main/java/com/github/onsdigital/babbage/search/helpers/SearchHelper.java

```diff
@@ -67,8 +74,31 @@ public static ONSSearchResponse search(ONSQuery queries) {
                 if (item.isFailure()) {
                     throw new ElasticsearchException(item.getFailureMessage());
                 }
-                helpers.add(resolveDetails(queries.get(i), new ONSSearchResponse(item.getResponse())));
-                i++;
+                ONSQuery query = queries.get(i);
+                ONSSearchResponse searchResponse = resolveDetails(query, new ONSSearchResponse(item.getResponse()));
+
+                List<ContentType> types = Arrays.asList(query.types());
+
+                // Check for retired product pages
+                if (query.size() == 1 &&
+                        types.contains(ContentType.product_page)) {
+                    // Get the hit - we only queried for 1 document so we should only get 1 back
+                    SearchHits searchHits = searchResponse.response.getHits();
+                    if (searchHits.getHits().length == 1) {
+                        SearchHit searchHit = searchHits.getAt(0);
+                        // ID is the url
+                        String url = searchHit.getId();
+                        if (!retiredUrls.contains(url)) {
+                            // OK to add response
+                            helpers.add(searchResponse);
+                            i++;
+                        }
+                    }
+                } else {
+                    // Not a topic match query, so go ahead and add the response as usual
+                    helpers.add(searchResponse);
+                    i++;
```

Comment:
What is `i`used for? I had a look but I can't see it being used other than the declaration and here - unless ive missed it